network/round-robin.go

```diff
@@ -0,0 +1,20 @@
+package network
+
+import "sync/atomic"
+
+type RoundRobin struct {
+	proxies []IProxy
+	next    atomic.Uint32
+}
+
+func NewRoundRobin(server *Server) *RoundRobin {
+	return &RoundRobin{proxies: server.Proxies}
+}
+
+func (r *RoundRobin) GetNextProxy() IProxy {
+	proxiesLen := uint32(len(r.proxies))
+
+	nextIndex := r.next.Add(1)
```

Comment:
Maybe check for overflows before increasing this indefinitely?
```suggestion
    if r.next.Load() >= math.MaxUint32 {
        r.next.Store(0)
    }
	nextIndex := r.next.Add(1)
```

Nit: maybe remove empty lines as well.

Reply:
the `sync/atomic` package can handle overflows by resetting the value to 0. To test this, added a new test case called `TestNextProxyOverflow`. also added error handling for `proxiesLen` and changed the return interface signature to `(IProxy, *gerr.GatewayDError)`